app/routes/submitted_orders.js

```diff
@@ -0,0 +1,54 @@
+import Ember from "ember";
+import AuthorizeRoute from "./authorize";
+
+export default AuthorizeRoute.extend({
+  orderService: Ember.inject.service(),
+
+  async beforeModel() {
+    const hasSubmittedOrders = await this.get(
+      "orderService"
+    ).hasSubmittedOrders();
+
+    if (!hasSubmittedOrders) {
+      let lastDraftOrder = await this.get("orderService").getLastDraft({
+        appointment: false
+      });
+      const orderId =
```

Comment:
`lastDraftOrder` is a single model, so it should not have a `length`
I think a `const orderId = lastDraftOrder && lastDraftOrder.get('id')` works here

Reply:
Done.